sdk/provisioning/Azure.Provisioning.CloudMachine/src/CDKLevel3/FeatureCollection.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Provisioning.CloudMachine;
+
+namespace Azure.CloudMachine;
+
+public class FeatureCollection
+{
+    private CloudMachineFeature[] _items = new CloudMachineFeature[4];
+    private int _count;
+
+    public bool TryFind<T>(out T? found) where T: CloudMachineFeature
+    {
+        for (int i = 0; i < _count; i++)
+        {
+            if (_items[i] is T item)
+            {
+                found = item;
+                return true;
+            }
+        }
+        found = default;
+        return false;
+    }
+
+    public void Add(CloudMachineFeature item)
```

Comment:
Is it intentional to allow duplicate feature types to be added? If so, we should account for that in `TryFind`

Reply:
I fixed TryFind